### PR TITLE
Add prepared statements and apis for calendar/profile page

### PR DESCRIPTION
**Pull request by dennywu84**

### CODE REVIEW

File: Backend/index.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const sql = require('mssql')
 const bcrypt = require('bcrypt');
 const session = require('express-session')
 const {calculateWeek} = require("./Calendar")
-const {pool, getCourseStatement, getReviewsStatement, addReviewStatement, createAccountStatement, loginAccountStatement} = require("./sql")
+const {pool, getCourseStatement, getReviewsStatement, addReviewStatement, createAccountStatement, loginAccountStatement, findPlannedClassesStatement, updateProfilePage} = require("./sql")
```
Comment (bytehut) [inline Backend/index.js]: should rename updateProfilePage to include 'Statement' for consistency

File: Backend/index.js [hunk 2]
```diff
@@ -145,9 +145,29 @@ app.post('/login', (req, res) => {
...
+    const email = req.session.userId
+    findPlannedClassesStatement.execute({findPlannedEmail: email}, (err, result) => {
+      if (err) {
+        res.status(500).send("Error encountered while retrieving courses. Please try again.")
+        console.log(err)
+        return
+      }
+      res.send(result.recordset)
+    })
+  } else {
```
Comment (bytehut) [inline Backend/index.js:163]: could return a 401 unauthorized status instead

File: Backend/index.js [hunk 3]
```diff
@@ -145,9 +145,29 @@ app.post('/login', (req, res) => {
...
+    res.status(500).send("Error encountered while retrieving courses. Please try again.")
+  }
+  res.sendFile(path.join(root, 'FrontEnd', 'DubSpotCalendar.html'))
```
Comment (bytehut) [inline Backend/index.js]: shouldnt send the html file here

File: Backend/index.js [hunk 4]
```diff
@@ -171,6 +191,15 @@ app.get('/Profile', (req, res) => {
   res.sendFile(path.join(root, 'FrontEnd', 'DubSpotProfile.html'))
 })
 
+app.get('/api/profile', (req, res) => {
+  if (req.session.userId !== undefined) {
+    const email = req.session.userId
+
+  }
+  res.sendFile(path.join(root, 'FrontEnd', 'DubSpotProfile.html'))
```
Comment (bytehut) [inline Backend/index.js]: shouldnt send the html file here

**Review by bytehut — CHANGES_REQUESTED** [review]
We also need a prepared statement to fetch the profile information, and an app.post that receives information from the client that calls the updateProfilePage

File: Backend/index.js (hunk 4 above)
Comment (dennywu84) [reply Backend/index.js]: Fixed

File: Backend/index.js (hunk 3 above)
Comment (dennywu84) [reply Backend/index.js]: Fixed

File: Backend/sql.js [hunk 5]
```diff
@@ -87,5 +95,6 @@ module.exports = {
   createAccountStatement,
   loginAccountStatement,
   findPlannedClassesStatement,
-  updateProfilePage
+  updateProfilePage,
```
Comment (bytehut) [inline Backend/sql.js]: should be updateProfilePageStatement

**Review by bytehut — CHANGES_REQUESTED** [review]

**Review by bytehut — APPROVED** [review]
Looks good! Thanks for getting back on the issues.

**Pull request merged**